cleanup-passes/typescript.ts

```diff
@@ -109,31 +131,35 @@ async function typescriptPass(element: ElementRepo): Promise<void> {
 
   const execOpts = {cwd: element.dir};
 
-  // Install the generator and its dependencies.
+  // Install the generator and its dependencies. Delete the package lock in case
+  // a newer version of the generator will change our types.
+  const packageLockPath = path.join(element.dir, 'package-lock.json');
+  if (await fse.pathExists(packageLockPath)) {
+    await fse.remove(packageLockPath);
+  }
   await execFilePromise('npm', ['install'], execOpts);
 
   // Run the generator (using the script we added above).
   await execFilePromise('npm', ['run', npmScriptName], execOpts);
 
-  let doCommit = false;
   const commitFiles = [];
   for (const changedFile of await element.repo.getStatus()) {
     const filepath = changedFile.path();
-    if (filepath.endsWith('.d.ts') || filepath === 'package.json' ||
-        filepath === '.travis.yml') {
-      doCommit = true;
-      commitFiles.push(filepath);
-    } else if (filepath === 'package-lock.json') {
-      // If all we did was update the package lock, don't bother with the
-      // commit. But do include it if we changed something else.
-      commitFiles.push(filepath);
+    if (filepath.endsWith('.d.ts')) {
+      updatedTypes = true;
+    } else if (filepath === '.travis.yml') {
+      updatedTravis = true;
+    } else if (
+        filepath === 'package.json' || filepath === 'package-lock.json') {
     } else {
       console.log(
           `${element.ghRepo.name}: Unexpected changed file: ${filepath}`);
+      continue;
     }
+    commitFiles.push(filepath);
   }
 
-  if (doCommit) {
+  if (updatedTypes || majorVersionBump || updatedNpmScript || updatedTravis) {
```

Comment:
Hm, this is pretty awkward. Is there a better way? What about `commitFiles.length > 0`?

Reply:
I don't want to commit if we e.g. only bumped the package lock. The idea is you should be able to run the script and it only makes PRs when there was a meaningful change. Similar logic existed before, it's just that I was setting the `doCommit` flag before, whereas now it's explicit what the conditions are at the end.

Reply:
Not blocking, but I kinda like having one purpose-built variable rather than four.

I sorta think that both solutions are a sign that we don't quite have this API right.

Reply:
Done.